packages/generate-text/src/index.ts

```diff
@@ -28,21 +28,25 @@ export interface GenerateTextResult {
   usage: GenerateTextResponseUsage
 }
 
-export const generateText = async (options: GenerateTextOptions): Promise<GenerateTextResult> =>
-  await chatCompletion({
-    ...options,
-    stream: false,
-  })
-    .then(async (res) => {
-      if (!res.ok) {
-        const error = await res.text()
-        throw new Error(`Error(${res.status}): ${error}`)
-      }
-      else {
-        return res.json() as Promise<GenerateTextResponse>
-      }
+export const generateText = async (options: GenerateTextOptions): Promise<GenerateTextResult> => {
+  const maxRoundTrip = options.maxRoundTrip ?? 10
+  let roundTrip = 0
+
+  while (roundTrip < maxRoundTrip) {
```

Comment:
As an alternative, maybe consider adding an internal `currentStep` variable that could be passed to the recursive function.

```ts
return await generateText({
  ...options,
  currentStep: currentStep + 1,
  ...overrides,
})
```